### PR TITLE
lift-mysql-mds

### PR DESCRIPTION
**Pull request by Soumyadeep-Oracle**

### CODE REVIEW

**Review by AlexanderHodicke — CHANGES_REQUESTED** [review]
@oheimburger Please review

File: cloud-architecture/custom-apps-and-consolidation/mysql-mds/files/locals.tf [hunk 1]
```diff
@@ -0,0 +1,31 @@
+# Copyright (c) 2023, Oracle and/or its affiliates.
+# Licensed under the Universal Permissive License v1.0 as shown at https://oss.oracle.com/licenses/upl.
+# Purpose: terraform configuration file to declare local variables.
+# @author: Vijay Kokatnur
+
+locals {
+  all_protocols = "all"
+
+  anywhere = "0.0.0.0/0"
```
Comment (oheimburger) [inline cloud-architecture/custom-apps-and-consolidation/mysql-mds/files/locals.tf:9]: Bad security design.

File: cloud-architecture/custom-apps-and-consolidation/mysql-mds/files/modules/network/security.tf [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+  egress_security_rules {
+    protocol    = local.all_protocols
+    destination = local.anywhere
+  }
+
+  ingress_security_rules {
+    #Required
+    protocol = local.tcp_protocol
+    source = "0.0.0.0/0"
```
Comment (oheimburger) [inline cloud-architecture/custom-apps-and-consolidation/mysql-mds/files/modules/network/security.tf:20]: Fixed valued. Use a local variable for that. Don't use local.anywhere!

File: cloud-architecture/custom-apps-and-consolidation/mysql-mds/files/modules/network/security.tf [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+        #Optional
+        max = "3306"
+        min = "3306"
+    }
+  }
+
+  ingress_security_rules {
+    #Required
+    protocol = local.tcp_protocol
+    source = "0.0.0.0/0"
```
Comment (oheimburger) [inline cloud-architecture/custom-apps-and-consolidation/mysql-mds/files/modules/network/security.tf:52]: Fixed valued. Use a local variable for that. Don't use local.anywhere!

File: cloud-architecture/custom-apps-and-consolidation/mysql-mds/files/modules/network/security.tf [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+    #Optional
+    description = "ingress port 33060 for MySQL DB"
+    source_type = "CIDR_BLOCK"
+    stateless = false
+    tcp_options {
+        #Optional
+        max = "33060"
```
Comment (oheimburger) [inline cloud-architecture/custom-apps-and-consolidation/mysql-mds/files/modules/network/security.tf:60]: Fixed valued. Use a local variable for that.

File: cloud-architecture/custom-apps-and-consolidation/mysql-mds/files/modules/network/security.tf [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+        min = "33060"
```
Comment (oheimburger) [inline cloud-architecture/custom-apps-and-consolidation/mysql-mds/files/modules/network/security.tf:61]: Fixed valued. Use a local variable for that.

File: cloud-architecture/custom-apps-and-consolidation/mysql-mds/files/terraform.tfvars [hunk 6]
```diff
@@ -0,0 +1,85 @@
+# Copyright (c) 2023, Oracle and/or its affiliates.
+# Licensed under the Universal Permissive License v1.0 as shown at https://oss.oracle.com/licenses/upl.
+# Purpose: terraform variable definition file.
+# @author: Vijay Kokatnur
+
+# Identity and access parameters
+home_region     = "eu-frankfurt-1"
```
Comment (oheimburger) [inline cloud-architecture/custom-apps-and-consolidation/mysql-mds/files/terraform.tfvars:7]: Use placeholder!

File: cloud-architecture/custom-apps-and-consolidation/mysql-mds/files/terraform.tfvars [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+region          = "eu-frankfurt-1"
```
Comment (oheimburger) [inline cloud-architecture/custom-apps-and-consolidation/mysql-mds/files/terraform.tfvars:8]: Use placeholder!

File: cloud-architecture/custom-apps-and-consolidation/mysql-mds/files/terraform.tfvars [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+
+
+# general oci parameters
+compartment_id = "ocid1.compartment.oc1.."
+label_prefix   = "none"
+
+# ssh keys
+ssh_private_key_path = "keys/id_rsa"
```
Comment (oheimburger) [inline cloud-architecture/custom-apps-and-consolidation/mysql-mds/files/terraform.tfvars:16]: Use placeholder.

File: cloud-architecture/custom-apps-and-consolidation/mysql-mds/files/terraform.tfvars [hunk 9]
```diff
@@ -0,0 +1,85 @@
...
+# ssh_public_key       = ""
+# ssh_public_key_path  = "none"
+ssh_public_key_path = "keys/id_rsa.pub"
```
Comment (oheimburger) [inline cloud-architecture/custom-apps-and-consolidation/mysql-mds/files/terraform.tfvars:19]: Use placeholder.

File: cloud-architecture/custom-apps-and-consolidation/mysql-mds/files/terraform.tfvars [hunk 10]
```diff
@@ -0,0 +1,85 @@
...
+
+#networking parameters
+subnet_id     = "ocid1.subnet.oc1..."
```
Comment (oheimburger) [inline cloud-architecture/custom-apps-and-consolidation/mysql-mds/files/terraform.tfvars:22]: Use placeholder.

File: cloud-architecture/custom-apps-and-consolidation/mysql-mds/files/terraform.tfvars [hunk 11]
```diff
@@ -0,0 +1,85 @@
...
+# netnum  = 2
+#vcn_id        = ""
+#ig_route_id   = ""
+
+assign_public_ip = false
+enable_ext      = false
+
+# Compute Instance
+instance_count = 3
+image_id    = "ocid1.image.oc1..."
```
Comment (oheimburger) [inline cloud-architecture/custom-apps-and-consolidation/mysql-mds/files/terraform.tfvars:36]: Use placeholder.

File: cloud-architecture/custom-apps-and-consolidation/mysql-mds/files/terraform.tfvars [hunk 12]
```diff
@@ -0,0 +1,85 @@
...
+}
+state    = "RUNNING"
+timezone = "Etc/UTC"
+type     = "public"
+#upgrade_bastion  = false
+
+#MySQL DB System Paramters
+mysql_shape_name  = "MySQL.VM.Standard2.4.60GB"
+
+mysql_db_subnet  = "ocid1.subnet.oc1..."
```
Comment (oheimburger) [inline cloud-architecture/custom-apps-and-consolidation/mysql-mds/files/terraform.tfvars:52]: Use placeholder.

File: cloud-architecture/custom-apps-and-consolidation/mysql-mds/files/terraform.tfvars [hunk 13]
```diff
@@ -0,0 +1,85 @@
...
+
+mysql_db_system_admin_username  = "admin" 
+
+mysql_db_system_admin_password  = "admin123"
```
Comment (oheimburger) [inline cloud-architecture/custom-apps-and-consolidation/mysql-mds/files/terraform.tfvars:56]: NO PASSWORD IN CODE!!!

File: cloud-architecture/custom-apps-and-consolidation/mysql-mds/files/terraform.tfvars
Comment (oheimburger) [inline cloud-architecture/custom-apps-and-consolidation/mysql-mds/files/terraform.tfvars:1]: Use placeholders. No real values!

**Review by oheimburger — CHANGES_REQUESTED** [review]
THIS BRANCH NEEDS TO BE REMOVED!

It contains user credentials and violates Oracle policy!

For correction:
- Make sure to follow proper English grammar and punctuation (like every sentence starts with a capital letter and ends with period or related punctuation).

File: cloud-architecture/custom-apps-and-consolidation/mysql-mds/files/terraform.tfvars (hunk 10 above)
Comment (Soumyadeep-Oracle) [reply cloud-architecture/custom-apps-and-consolidation/mysql-mds/files/terraform.tfvars:22]: subnet_id     = "ocid1.subnet.oc1..." This is already a placeholder. Any other suggested format?

File: cloud-architecture/custom-apps-and-consolidation/mysql-mds/files/terraform.tfvars (hunk 13 above)
Comment (Soumyadeep-Oracle) [reply cloud-architecture/custom-apps-and-consolidation/mysql-mds/files/terraform.tfvars:56]: Will this format work - for eg:   username=<USERNAME> , password=<PASSWORD>

**Review by oheimburger — CHANGES_REQUESTED** [review]
Please resolve pending changes.

**Comment by AlexanderHodicke** [comment]
@Soumyadeep-Oracle Please fi the feedback, or close the PR. TY

**Comment by AlexanderHodicke** [comment]
@oheimburger "THIS BRANCH NEEDS TO BE REMOVED!" Do we need to close the PR and delete the branch? Why don't you delete it right away?

**Comment by oheimburger** [comment]
There are some unresolved comments for this branch. Please resolve them.

**Comment by oheimburger** [comment]
@AlexanderHodicke  can't delete this branch.

**Pull request closed without merging**